Crypto101.org

```diff
@@ -2912,7 +2912,40 @@ http://www.cs.ucdavis.edu/~rogaway/papers/relates.pdf for further reading
 
 *** MD5
 
-TODO: Explain MD5
+MD5 is a hash function designed by Ronald Rivest in 1991 as an extension of MD4. This hash function outputs 128-bit digests. Over the course of the years, the cryptographic community has repeatedly uncovered MD5's weaknesses. In 1993, Bert den Boer and Antoon Bosselaers published a paper demonstrating "pseudo-collisions" for the compression function of MD5. Dobbertin expanded upon this research and was able to produce collisions for the compression function. In 2004, based on Dobbertin's work, Xiaoyun Wang, Dengguo Feng, Xuejia Lai and Hongbo Yu showed that MD5 is vulnerable to real collision attacks. The last straw came when Xiaoyun Wang et al. managed to generate colliding X.509 certificates and then presented a distinguishing attack on HMAC-MD5.
+
+Due to all the issues listed above it is not recommended to use MD5 for its intended purpose of generating digital signatures.
```

Comment:
I'd remove "intended purpose of" -- MD5 is a hash function, and it's intended for more than signature generation; also, please add that while it's not recommended for new implementations, HMAC-MD5 is still a secure form of message authentication. (I mostly just want to hammer that collision resistance, pre-image resistance and 2nd pre-image resistance are not the same and you can't just go ahead and say "MD5 is insecure!!1" -- I've seen people yell that noise about HMAC-MD5, unfortunately.)